site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,666 @@
+                cy.get('#late_days').clear();
+                cy.get('#late_days').type(days);
+                cy.get('input[type=submit]').click();
+                cy.wait(3000);
+                late_days_remaining[user_id] = days;
+            }
+        });
+
+        it('should make bulk late days has been emptied out', () => {
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.login('instructor');
+
+            for (const test_case of test_info) {
+                const { g_id, g_title } = test_case;
+                const late_users = all_late_users[g_id];
+
+                for (const user_id in late_users) {
+                    // Gradeable # of late days used should be empty
+                    cy.get(`[data-user="${user_id}"] > [data-before-content="${g_title}"]`)
+                        .then((cell) => expect(cell.text().trim()).to.equal(''));
+
+                    // Remaining late days isnt known, should be empty
+                    cy.get(`[data-user="${user_id}"] > [data-before-content="Late Days Remaining"]`)
+                        .then((cell) => expect(cell.text().trim()).to.equal(''));
+                }
+            }
+        });
+
+        it('should make sure late day status has updated', () => {
+            for (const test_case of test_info) {
+                const { g_id, g_title, days_allowed } = test_case;
+                const late_users = all_late_users[g_id];
+
+                for (const user_id in late_users) {
+                    cy.visit([]);
+                    cy.login(user_id);
+                    cy.visit(['sample', 'late_table']);
+
+                    // Wait for login change to take place
+                    cy.wait(3000);
+                    const status = predictedStatus(days_allowed, late_users[user_id], late_days_remaining[user_id]);
+
+                    // Find late day status within the row in the late day usage table
+                    cy.get('td[data-before-content="Event/Assignment"]')
+                        .contains(g_title)
+                        .siblings('td[data-before-content="Status"]')
+                        .contains(status)
+                        .should('exist');
+
+                    // Update expected late days remaining
+                    if (status === 'Late') {
+                        late_days_remaining[user_id] -= late_users[user_id];
+                    }
+
+                    cy.logout();
+                }
+            }
+        });
+
+        it('should remove late day cache after deletion of late days', () => {
+            cy.visit(['sample', 'late_days']);
+            cy.login('instructor');
+
+            const deleteLateDays = () => {
+                cy.get('div.content').then((table) => {
+                    if (table.find('td[data-before-content="Delete"]').length > 0) {
+                        cy.wrap(table).find('td[data-before-content="Delete"]').first().click();
+                        cy.wait(3000);
```

Comment:
Instead of explicitly waiting, is there an element that should/shouldn't exist after the delete has happened? If you cy.get('element').should('not.exist'), it has a timeout of 5s, but will continue as soon as it is not visible, instead of waiting 3s every time.